### PR TITLE
Remove System.Reflection dependencies.

### PR DESCRIPTION
**Pull request by serkantkaraca**
UWP doesn't support most of the System.Reflection library. We are removing this dependency and reimplementing management link.

Addressing issue https://github.com/Azure/azure-event-hubs-dotnet/issues/47

\+ Fixing flaky ReceiveTimeout CIT.

### CODE REVIEW

File: src/Microsoft.Azure.EventHubs/Amqp/AmqpEventHubClient.cs [hunk 1]
```diff
@@ -120,14 +119,13 @@ protected override Task OnCloseAsync()
             try
             {
                 var timeoutHelper = new TimeoutHelper(this.ConnectionStringBuilder.OperationTimeout);
-                string serviceClientAddress = AmqpClientConstants.ManagementAddress;
-
-                string entityType = AmqpClientConstants.ManagementEventHubEntityTypeValue;
-                SecurityToken token = await this.TokenProvider.GetTokenAsync(this.ConnectionStringBuilder.Endpoint.AbsoluteUri, ClaimConstants.Manage, timeoutHelper.RemainingTime()).ConfigureAwait(false);
+                SecurityToken token = await this.TokenProvider.GetTokenAsync(
+                    this.ConnectionStringBuilder.Endpoint.AbsoluteUri, 
+                    ClaimConstants.Manage, timeoutHelper.RemainingTime());
```
Comment (jtaubensee) [inline src/Microsoft.Azure.EventHubs/Amqp/AmqpEventHubClient.cs]: Add ConfigureAwait(false)?

File: src/Microsoft.Azure.EventHubs/Primitives/ClientInfo.cs [hunk 2]
```diff
@@ -25,7 +25,9 @@ static ClientInfo()
                 framework = GetAssemblyAttributeValue<TargetFrameworkAttribute>(assembly, f => f.FrameworkName);
 #if NETSTANDARD1_3
                 platform = System.Runtime.InteropServices.RuntimeInformation.OSDescription;
-#else
+#elif UAP
+                platform = null;
```
Comment (xinchen10) [inline src/Microsoft.Azure.EventHubs/Primitives/ClientInfo.cs]: Use a constant here like "uap".

File: src/Microsoft.Azure.EventHubs/project.json [hunk 3]
```diff
@@ -25,14 +25,20 @@
   },
 
   "frameworks": {
+    "uap10.0": {
+      "dependencies": {
+        "Microsoft.NETCore.UniversalWindowsPlatform": "5.2.2"
+      },
+      "buildOptions": {
+        "define": [ "UAP" ]
```
Comment (xinchen10) [inline src/Microsoft.Azure.EventHubs/project.json]: Not needed. "UAP10_0" should be automatically defined.

File: src/Microsoft.Azure.EventHubs/Amqp/Management/AmqpServiceClient.cs [hunk 4]
```diff
@@ -13,359 +13,96 @@ namespace Microsoft.Azure.EventHubs.Amqp.Management
...
-                {
-                    this.SetException(exception);
-                }
-                else
-                {
-                    this.SetResult((TResult)result);
-                }
-            }
+            ActiveClientRequestResponseLink activeClientLink = await this.eventHubClient.OpenRequestResponseLinkAsync(
+                "svc", this.Address, null, AmqpServiceClient.RequiredClaims, timeout);
```
Comment (xinchen10) [inline src/Microsoft.Azure.EventHubs/Amqp/Management/AmqpServiceClient.cs]: ConfigureAwait(false)

**Pull request merged**